presto-record-decoder/src/main/java/io/prestosql/decoder/json/AbstractDateTimeJsonValueProvider.java

```diff
@@ -70,10 +70,10 @@ public final long getLong()
         if (type.equals(TIME)) {
             return millis * PICOSECONDS_PER_MILLISECOND;
         }
-        if (type.equals(TIMESTAMP)) {
+        if (type instanceof TimestampType) {
             return millis;
         }
-        if (type.equals(TIMESTAMP_WITH_TIME_ZONE) || type.equals(TIME_WITH_TIME_ZONE)) {
+        if (type instanceof TimestampWithTimeZoneType || type.equals(TIME_WITH_TIME_ZONE)) {
```

Comment:
same

Reply:
in this case, the encoding is:
* p <= 3, a long containing the milliseconds since the epoch packed with the time zone id
* p > 3, an instance of LongTimestampWithTimeZone